python/ray/dag/compiled_dag_node.py

```diff
@@ -296,12 +298,19 @@ def __init__(
                 do not support binding kwargs to other DAG nodes, so the values
                 of the dictionary cannot be Channels.
         """
+        from ray.dag import CollectiveOutputNode
```

Comment:
Can you try to find what the circular import is?

Reply:
`dag_node.py` imports `build_compiled_dag_from_ray_dag` from `compiled_dag_node.py`,
`compiled_dag_node.py` imports `CollectiveOutputNode` from `collective_node.py`, and
`collective_node.py` imports `DAGNode` from `dag_node.py`.